### PR TITLE
added "jpg or png" option

### PR DESCRIPTION
**Pull request by tailoric**
Let's the user decide if they either want to upload a .jpg or a .png image

http://i.imgur.com/AzTLciH.png

### CODE REVIEW

File: imgur-flatten-upload.py [hunk 1]
```diff
@@ -44,8 +44,12 @@ def upload_to_imgur(image, layer):
     try:
         # Save as PNG
         outputFolder = tempfile.gettempdir()
-        f = os.path.join(outputFolder, "imgur.png")
-        gimp.pdb.file_png_save(image, image.flatten(), f, "raw_filename", 0, 9, 0, 0, 0, 0, 0)
+        if file_type == 0:
+            f = os.path.join(outputFolder,layer.name + ".jpg")
+            gimp.pdb.file_jpeg_save(image, layer, f, "raw_filename", 0.85, 0, 0, 0, 0, 0,0,0,0)
```
Comment (utecht) [inline imgur-flatten-upload.py]: Should be image.flatten() instead of layer as the second argument.

File: imgur-flatten-upload.py [hunk 2]
```diff
@@ -44,8 +44,12 @@ def upload_to_imgur(image, layer):
...
+        else :
+            f = os.path.join(outputFolder, layer.name + ".png")
+            gimp.pdb.file_png_save(image, layer, f, "raw_filename", 0, 9, 0, 0, 0, 0, 0)
```
Comment (utecht) [inline imgur-flatten-upload.py]: Same as above.

**Review by utecht — CHANGES_REQUESTED** [review]
At the very least these need to be changed to fix the flattening, however see my main comment about an alternative idea.

**Comment by utecht** [comment]
I like this idea but am not the biggest fan of needing a dialog box every time to select the format.  With the modular nature of gimp plugins however, it is just easier to add a 3rd script to accommodate additional options.

You could make imgur-options.py where you could select png/jpg from a dropdown and have a checkbox for flattening the image out.  Also if any additional options come up in the future they could be added there.

Thanks for the pull request.

File: imgur-flatten-upload.py (hunk 1 above)
Comment (tailoric) [reply imgur-flatten-upload.py]: ups, sorry an overlooked copy paste mistake.

I'll fix it.

**Review by tailoric — COMMENTED** [review]
Well not exactly a checkbox, it's more a button that says either "Yes" or "No"

**Comment by utecht** [comment]
Great this all looks good.  I'll go ahead and merge it in.

Out of curiosity where did you find out about this?  I didn't think anyone else used this.

**Comment by tailoric** [comment]
I just googled for "gimp imgur upload" and this was the first result.

After that I just looked up a tutorial at how to do user inputs, because I wanted to have the option of choosing the filetype.

And I thought that it would be a good idea to have it in the original repository, because it looks like [I'm not the only one who wanted the option](https://github.com/tschuy/gimp-export-to-imgur/issues/2)

**Pull request merged**